src/commonTest/resources/dev/rhovas/interpreter/programs/rosettacode/Fibonacci.rho

```diff
@@ -0,0 +1,26 @@
+func pfib(n: Integer): Integer {
+    require n >= 0;
+
+    var lastFib = -1;
+    var fib = 1;
```

Comment:
Can we rename these `previous` and `current`? I think we should also shift this to `1`, `0` and cut an iteration out of the next loop by starting at `1`, it took me a bit to figure out exactly what was happening here and why this worked.

Reply:
Yeah the previous = -1  and current = 1 convention was one I saw other solutions using to resolve the n = 0 edge case. Of course, that could also be resolved by a guard clause for n = 0, but that felt slightly arbitrary. If I put in a guard clause for 0, why not for 1 too? That would *technically* improve performance by reducing the number of loops. Heck, why not have a guard clause for every possible Fibonacci number from negative infinity to positive infinity. Easy solution.

Starting with previous=1, current=0, and removing one loop does resolve this slightly more efficiently so I'll go with that.

Reply:
It's mainly to avoid questions of unknown behavior down the road, though performance is relevant as well. Currently, since everything is interpreted it uses dynamic scope to Do The Right Thing. However, once this becomes compiled there are a lot of challenges related to closures and variable capturing (e.g. https://news.ycombinator.com/item?id=33160236). Those are hard problems to solve, so until I figure out what happens there it's better to avoid reliance on behavior there.

I have some ideas for annotations like `@Immediate` to indicate a lambda can only be run with the function it was passed to (and not stored somewhere to run later), but that comes *way* down the line.

> Heck, why not have a guard clause for every possible Fibonacci number from negative infinity to positive infinity. Easy solution.

Welcome to caching, one of the 2* hard problems in computer science ;)

Reply:
Moved my answer for lambdas/performance over to https://github.com/Rhovas/Interpreter/pull/12#discussion_r1014917890

> Heck, why not have a guard clause for every possible Fibonacci number from negative infinity to positive infinity. Easy solution.

Welcome to caching, one of the 2* hard problems in computer science ;)